src/libopensc/pkcs15-cac.c

```diff
@@ -352,12 +352,14 @@ static int sc_pkcs15emu_cac_init(sc_pkcs15_card_t *p15card)
 		r =  sc_pkcs15_read_certificate(p15card, &cert_info, &cert_out);
 		if (r < 0 || cert_out->key == NULL) {
 			sc_debug(card->ctx, SC_LOG_DEBUG_NORMAL, "Failed to read/parse the certificate r=%d",r);
+			sc_pkcs15_free_certificate(cert_out);
```

Comment:
if `cert_out` is `NULL`, then we will hit an `assert` in `sc_pkcs15_free_certificate`. So you should check for NULL here or remove the assert.